### PR TITLE
huge update (using spawn , avoid sync functions, unicode fixes)

### PR DESCRIPTION
**Pull request by digital-flowers**
Ok the following is a really huge update and fully tested in OSX and Windows :
1- **Child Processes** 
recently i discovered that i have a lot of crashes because of mediainfo library and i think one of the biggest issues was using child_process.exec where it was not design to return huge data we should use spawn, please read this: 
screen shot 2016-10-06 at 4 37 48 pm

to solve this i have created a nodejs module called [fluent-child-process](https://github.com/digital-flowers/fluent-child-process) this module handle calling child process with this advantages
- it handle many cases to make sure the nodejs main process not crashing
- you don't have to care about character escaping and creating safe OS paths  since it is handled internally by node.js
- no need for max buffer size since i am using child_process.spawn

2- **Glob Sync** 
as you know it is always better to avoid using sync functions and i have found that you used glob.sync function to resolve global files pattern so i have replace it with Promises.all and Promises.reduce to make sure the main process not hanging 

**Note** i have tested the change in both OSX/ macOS Sierra/Windows also UTF-8/UTF-16le file paths also i have tested more than 5000 media file with filenames that has special characters and unicode names (Korean/English/Arabic), and also extracting embedded cover photos from 3000 mp3 files ;)

waiting for your review :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,12 +22,12 @@ mi('foo/bar.mkv', 'foo/bar2.avi').then(function(data) {
...
-### Using child_process power
-
-You can pass an object as first argument to use exec options. See [Node child_process](https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback).
+### Using fluent-child-process power
+This library built over [fluent-child-process](https://github.com/digital-flowers/fluent-child-process) so still you can pass any extra options to fluent-child-process 
+You can pass an object as first argument to use fluent-child-process options. See [fluent-child-process](https://github.com/digital-flowers/fluent-child-process).
 
 ```js
-require('mediainfo-wrapper')({maxBuffer: 'infinity'}, 'foo/bar.mkv', 'foo/bar2.avi').then...
+require('fluent-mediainfo')({logger:mylogger}, 'foo/bar.mkv', 'foo/bar2.avi').then...
```
Comment (vankasteelj) [inline README.md:30]: mediainfo-wrapper is the name of the npm module, you must have forgotten to reverse it

File: package.json [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
-        "media",
-        "metadata",
-        "codecs"
-    ],
-    "author": {
-        "name": "Jean van Kasteel",
-        "email": "vankasteelj@gmail.com"
-    }
+  "name": "mediainfo-wrapper",
+  "version": "1.3.1",
```
Comment (vankasteelj) [inline package.json:3]: From 1.2.0, you need to update to 2.0.0 because it's incompatible with v1 (options in first object change, so ppl need to update their code = breaking change, see http://semver.org/)

**Review by vankasteelj — CHANGES_REQUESTED** [review]
I need to test on linux first, but it does seem to work great

File: README.md (hunk 1 above)
Comment (digital-flowers) [reply README.md:30]: ooh sorry i was thinking about forking your module to something called "fluent-mediainfo" then i changed my mind :P

File: package.json (hunk 2 above)
Comment (digital-flowers) [reply package.json:3]: yes i agree it should be version 2,  please test it in linux and please let me know what do you think :)

**Comment by vankasteelj** [comment]
It doesn't work on Linux. `Uncaught TypeError: Cannot read property 'get' of undefined => mediainfo-wrapper/node_modules/fluent-child-process/lib/processor.js (215)`

**Comment by digital-flowers** [comment]
sorry i will test it again under linux and get back to you :)

**Comment by vankasteelj** [comment]
mmmmmh
